src/Entity/Event.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use JMS\Serializer\Annotation as Serializer;
+use Sulu\Bundle\MediaBundle\Entity\MediaInterface;
+
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="app_event")
+ * @Serializer\ExclusionPolicy("all")
+ */
+class Event
+{
+    public const RESOURCE_KEY = 'events';
+    public const FORM_KEY = 'event_details';
+    public const LIST_KEY = 'events';
+    public const SECURITY_CONTEXT = 'sulu.events.events';
+
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     *
+     * @Serializer\Expose()
+     */
+    private ?int $id = null;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     *
+     * @Serializer\Expose()
+     */
+    private string $name;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=MediaInterface::class)
+     * @ORM\JoinColumn(onDelete="SET NULL")
+     */
+    private ?MediaInterface $image = null;
+
+    /**
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     *
+     * @Serializer\Expose()
+     */
+    private ?\DateTimeImmutable $startDate;
+
+    /**
+     * @ORM\Column(type="datetime_immutable", nullable=true)
+     *
+     * @Serializer\Expose()
+     */
+    private ?\DateTimeImmutable $endDate;
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getName(): string
+    {
+        return $this->name ?? '';
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    public function getImage(): ?MediaInterface
+    {
+        return $this->image;
+    }
+
+    /**
+     * @return array<string, mixed>
+     *
+     * @Serializer\VirtualProperty()
+     * @Serializer\SerializedName("image")
+     */
+    public function getImageData(): ?array
+    {
+        if ($image = $this->getImage()) {
+            return [
+                'id' => $image->getId(),
+            ];
+        }
+
+        return null;
+    }
```

Comment:
The Sulu form view expects the API to return the data in the same format as it is sent to the server when submitting the form. To do this, we can use a `VirtualProperty` method that returns the expected data instead of exposing the `Event::image` property directly.